ui/org.eclipse.pde.core/META-INF/MANIFEST.MF

```diff
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %name
 Bundle-SymbolicName: org.eclipse.pde.core; singleton:=true
-Bundle-Version: 3.17.300.qualifier
+Bundle-Version: 3.17.400.qualifier
```

Comment:
Why have you changed it? This causes now API error because it was already changes for the 4.31 release.

Please consider quickly revert this before next IBuild kicks in.

Reply:
I've set https://github.com/eclipse-pde/eclipse.pde/pull/948

Reply:
> Why have you changed it? This causes now API error because it was already changes for the 4.31 release.

Why it not failed this PR?

Reply:
I guess this is yet another API tools mystery.

Reply:
There was another issue with API error not being reported, see https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/914 .
Note that https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/914 is reported in SDK build and in the IDE.
So it could be some new tycho build issue.

Reply:
> There was another issue with API error not being reported

The green build reports it as a **warning**

> [2023-11-29T07:02:13.259Z] [WARNING] [API WARNING] File MANIFEST.MF at line 5: The service version is increased unnecessarily since either the major or minor or service version is already increased (location: /home/jenkins/agent/workspace/eclipse.pde_PR-940/ui/org.eclipse.pde.core/META-INF/MANIFEST.MF)

> So it could be some new tycho build issue.

API tools is provided by PDE not Tycho, Tycho just provides a wrapper around it, so either it is configured differently `WARNING` vs `ERROR` or the configuration is not picket up by Tycho @vik-chand might better know if/how/where this is to be configured.

Reply:
> Tycho just provides a wrapper around it

That is what I've meant with SDK/IDE build showing error.

I have no idea if it is the wrapper or the tycho config or whatever else. 
API tooling in the IDE works.

Reply:
I have now debugged the case and the problem is that `ApiProblem.getSeverity()` **always** return the default as it passes `null` as a project... Why it works in IDE/SDK ist because here Error Markers are used and when ApiTools converts `ApiProblem` > `Error Marker` it does _IGNORE_ `ApiProblem.getSeverity()` and instead computes the value again but this time using current build project... will replicate this behaviour with Tycho as well ...

Reply:
Actually ApiProblem.getSeverity() is only used in two places:

1. ApiProblem.toString (what will print wrong value then or at least ignores the project settings)
2. In a `org.eclipse.pde.api.tools.problems.tests.ApiProblemTests.testGetSeverity() `

so actually I think that method can completely be deleted to prevent further confusion...